### PR TITLE
Support HDFS in BucketUtils file operations.

### PR DESCRIPTION
**Pull request by tomwhite**

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/dev/pipelines/bqsr/BaseRecalibratorDataflowUtils.java [hunk 1]
```diff
@@ -92,11 +90,11 @@
...
         for (String fn : getRelatedFiles(filename)) {
-            if (BucketUtils.isCloudStorageUrl(fn)) {
-                // make sure we can access those files on Google Cloud Storage.
-                GcsPath path = GcsPath.fromUri(fn);
+            if (BucketUtils.isRemoteStorageUrl(fn)) {
+                // make sure we can access those remote files.
+                InputStream inputStream = BucketUtils.openFile(fn, popts);
                 // this will throw if we can't get to the file.
-                new GcsUtil.GcsUtilFactory().create(popts).fileSize(path);
+                int ignored = inputStream.read();
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/dev/pipelines/bqsr/BaseRecalibratorDataflowUtils.java]: Use try-with-resources to ensure that this `InputStream` is closed.

File: src/main/java/org/broadinstitute/hellbender/dev/tools/walkers/bqsr/BaseRecalibratorDataflow.java [hunk 2]
```diff
@@ -183,12 +181,10 @@ protected static String pickOutputTablesPath(boolean remote, String stagingLocat
...
-        if (BucketUtils.isCloudStorageUrl(beforePath)) {
+        if (BucketUtils.isRemoteStorageUrl(beforePath)) {
             // set up ingestion on the cloud
             // but read the header locally
-            GcsPath path = GcsPath.fromUri(beforePath);
-            InputStream inputstream = Channels.newInputStream(new GcsUtil.GcsUtilFactory().create(pipeline.getOptions())
-                    .open(path));
+            InputStream inputstream = BucketUtils.openFile(beforePath, pipeline.getOptions());
             SamReader reader = SamReaderFactory.makeDefault().validationStringency(ValidationStringency.SILENT).open(SamInputResource.of(inputstream));
             header = reader.getFileHeader();
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/dev/tools/walkers/bqsr/BaseRecalibratorDataflow.java]: Insert a call to `reader.close()` after this line.

File: src/test/java/org/broadinstitute/hellbender/utils/dataflow/BucketUtilsTest.java [hunk 3]
```diff
@@ -68,4 +74,30 @@ public void testCopyAndDeleteGCS() throws IOException, GeneralSecurityException
...
+    @Test
+    public void testCopyAndDeleteHDFS() throws IOException, GeneralSecurityException {
+        final String src = publicTestDir + "empty.vcf";
+        File dest = createTempFile("copy-empty", ".vcf");
+
+        MiniDFSCluster cluster = null;
+        try {
+            cluster = new MiniDFSCluster.Builder(new Configuration()).build();
+            String staging = cluster.getFileSystem().getWorkingDirectory().toString();
+            final String intermediate = BucketUtils.randomRemotePath(staging, "test-copy-empty", ".vsf");
```
Comment (droazen) [inline src/test/java/org/broadinstitute/hellbender/utils/dataflow/BucketUtilsTest.java]: Did you mean `.vsf`, or `.vcf`?

**Comment by droazen** [comment]
:+1: Looks good to me -- a couple of extremely minor comments to address, then should be ready to merge.

**Comment by tomwhite** [comment]
Thanks for the review @droazen. I've updated the PR with the fixes you suggested.

**Pull request merged**